Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -1212,6 +1212,7 @@ - (void)viewDidAppear:(BOOL)animated {
     }
     [self updateNavigationBarSubtitleLabel];
     [MarkIdentityAsSeenJob runWithThread:self.thread];
+    [ProfileFetcherJob runWithThread:self.thread networkManager:self.networkManager];
```

Comment:
I made this choice mainly to account for the fact that you might not leave a thread for a long time. 

e.g. you may background the app and come back to the conversation. 

Maybe I should move it to viewDidLoad and then again on UIApplicationWillEnterForegroundNotification

> ...When we add a member to a group?

That requires sending a message to that group member, so any SN changes have to be resolved *before* they can be added. Not really addressed by this IMO.